sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -791,53 +805,69 @@ private BasicDocumentRevision deleteDocumentInQueue(SQLDatabase db, final String
 
         CouchUtils.validateRevisionId(prevRevId);
 
-                    BasicDocumentRevision deletedRevision = null;
+        BasicDocumentRevision deletedRevision = null;
+            BasicDocumentRevision preRevision;
+        try {
+            preRevision = getDocumentInQueue(db, docId, prevRevId);
+        }  catch (AttachmentException e){
+            throw new DocumentNotFoundException(e);
+        }
```

Comment:
`createDocument` throws the attachment exception, to stop the datastore creating a document when it shouldn't error because it failed to load the parent https://github.com/cloudant/sync-android/blob/43599_improve_exceptions/sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java#L631